src/main/java/javax/jmdns/impl/ServiceTypeDecoder.java

```diff
@@ -40,7 +45,7 @@ static Map<Fields, String> decodeQualifiedNameMapForType(String type) {
             name = ServiceInfoImpl.removeSeparators(casePreservedType.substring(0, index));
             domain = casePreservedType.substring(index);
             application = "";
-        } else if ((!aType.contains("_")) && aType.contains(".")) {
+        } else if (recordType == DNSRecordType.TYPE_A || recordType == DNSRecordType.TYPE_AAAA || (!aType.contains("_")) && aType.contains(".")) {
```

Comment:
There should be no risk here, but I like to see these added checks inverted (like `DNSRecordType.TYPE_A == recordType`) because of the nullability of recordType. If anyone ever tries to convert this to a .equals operation in the future, it will not cause a NullPointerException that way.